src/http.rs

```diff
@@ -89,13 +184,22 @@ where
         tls_settings: impl Into<MaybeTlsSettings>,
         proxy_config: &ProxyConfig,
     ) -> Result<HttpClient<B>, HttpError> {
-        HttpClient::new_with_custom_client(tls_settings, proxy_config, &mut Client::builder())
+        HttpClient::new_with_custom_client(tls_settings, proxy_config, &mut Client::builder(), None)
+    }
+
+    pub fn new_asdasd(
```

Comment:
please don't look at names right now :) I just wanted to make it working end to end :)